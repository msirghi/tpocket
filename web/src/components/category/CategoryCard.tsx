import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';

interface IProps {
  name: string;
  description?: string;
  imageUrl?: string;
  onEdit: Function;
  onRemove: Function;
}

const useStyles = makeStyles({
  root: {
    minWidth: 345,
  },
  media: {
    height: 140,
  },
});

export const CategoryCard: React.FC<IProps> = ({name, imageUrl, onEdit, onRemove}) => {
  const classes = useStyles();

  return (
    <Card className={classes.root}>
      <CardActionArea>
        <CardMedia
          className={classes.media}
          image={imageUrl}
          title="Contemplative Reptile"
        />
        <CardContent>
          <Typography gutterBottom variant="h5" component="h2" data-test={'card-name'}>
            {name}
          </Typography>
          {/*<Typography variant="body2" color="textSecondary" component="p">*/}
          {/*  Lizards are a widespread group of squamate reptiles, with over 6,000 species, ranging*/}
          {/*  across all continents except Antarctica*/}
          {/*</Typography>*/}
        </CardContent>
      </CardActionArea>
      <CardActions>
        <Button size="small" color="primary" data-test={'card-edit'} onClick={() => onEdit()}>
          Edit
        </Button>
        <Button size="small" color="primary" data-test={'card-remove'} onClick={() => onRemove()}>
          Remove
        </Button>
      </CardActions>
    </Card>
  );
}
