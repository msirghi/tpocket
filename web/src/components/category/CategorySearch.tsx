import React from 'react';
import { Grid, TextField } from "@material-ui/core";
import CloseIcon from '@material-ui/icons/Close';
import Typography from '@material-ui/core/Typography';

interface IProps {
  onClose: Function;
  onChange: Function;
  resultCount: number;
  resultCountShown: boolean;
}

export const CategorySearch: React.FC<IProps> = ({ onClose, onChange, resultCount, resultCountShown }) => {
  return (
    <>
      <Grid container spacing={ 1 } alignItems="flex-end" justify={ 'center' } className={ 'pb-15' }>
        <Grid item>
          <TextField
            data-test={'category-name-field'}
            // autoFocus
            onChange={ (e) => onChange(e) }
            style={ { width: 300 } }
            label="Category name"
          />
        </Grid>
        <Grid item>
          <CloseIcon data-test={'close-icon'} onClick={ () => onClose() }/>
        </Grid>
      </Grid>
      { resultCountShown && resultCount >= 0 && <div className={ 'text-center' }>
        <Typography variant="subtitle1" gutterBottom data-test={'result-counter'}>
          { resultCount } results found.
        </Typography>
      </div> }
    </>
  );
}
