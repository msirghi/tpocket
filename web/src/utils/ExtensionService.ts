const monthNames: Array<string> = [
  'January',
  'February',
  'March',
  'April',
  'May',
  'June',
  'July',
  'August',
  'September',
  'October',
  'November',
  'December'
];

const getMonthName = (monthNumber: number) => monthNames[monthNumber];

export default {
  getMonthName
};
