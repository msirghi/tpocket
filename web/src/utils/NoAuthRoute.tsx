import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import { getAccessToken } from '../accessToken';

// @ts-ignore
export const NoAuthRoute = ({ Component, ...rest }) => (
  <Route
    {...rest}
    render={(props) => (getAccessToken() ? <Redirect to='/home' /> : <Component {...props} />)}
  />
);
