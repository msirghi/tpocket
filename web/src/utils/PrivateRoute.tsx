import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import { getAccessToken } from '../accessToken';

// @ts-ignore
export const PrivateRoute = ({ Component, ...rest }) => (
  <Route
    {...rest}
    render={(props) => (getAccessToken() ? <Component {...props} /> : <Redirect to='/login' />)}
  />
);
