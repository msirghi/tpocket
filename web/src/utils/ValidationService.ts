const passwordStrength = require('check-password-strength');

export const validateEmailStr = (email: string) => {
  const re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
  return re.test(String(email).toLowerCase());
};

export const isPasswordWeak = (password: string): boolean => {
  return passwordStrength(password).value === 'Weak';
};

const validateName = (name: string) => {
  const re = /^(?=.{2,50}$)[a-z]+(?:['_.\s][a-z]+)*$/i;
  return re.test(String(name).toLowerCase());
};

export default {
  isPasswordWeak,
  validateEmailStr,
  validateName
};
