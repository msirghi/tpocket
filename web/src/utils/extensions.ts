const passwordStrength = require('check-password-strength');

const monthNames: Array<string> = [
  'January',
  'February',
  'March',
  'April',
  'May',
  'June',
  'July',
  'August',
  'September',
  'October',
  'November',
  'December',
];

export const getMonthName = (monthNumber: number) => monthNames[monthNumber];

export const validateEmailStr = (email: string) => {
  return !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(email);
};

export const isPasswordWeak = (password: string): boolean => {
  return passwordStrength(password).value === 'Weak';
};
