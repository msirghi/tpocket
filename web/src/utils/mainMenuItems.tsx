import React from "react";
import { IMenuItem } from "../interfaces/IMenuItem";
import HomeIcon from '@material-ui/icons/Home';
import AccountCircleIcon from '@material-ui/icons/AccountCircle';
import CategoryIcon from '@material-ui/icons/Category';
import HelpOutlineIcon from '@material-ui/icons/Help';
import ExitToAppIcon from '@material-ui/icons/ExitToApp';
import EmailIcon from '@material-ui/icons/Email';

export const mainMenuItems: Array<IMenuItem> = [
  {
    id: 1,
    link: '/home',
    name: 'Home',
    icon: <HomeIcon className={ 'light-grey-icon' }/>
  },
  {
    id: 2,
    link: '/account',
    name: 'Account',
    icon: <AccountCircleIcon className={ 'light-grey-icon ' }/>
  },
  {
    id: 3,
    link: '/categories',
    name: 'Categories',
    icon: <CategoryIcon className={ 'light-grey-icon' }/>
  }
];

export const secondaryMenuItems: Array<IMenuItem> = [
  {
    id: 1,
    link: '/help',
    name: 'Help',
    icon: <HelpOutlineIcon className={ 'light-grey-icon' }/>
  },
  {
    id: 2,
    link: '/writeus',
    name: 'Support',
    icon: <EmailIcon className={ 'light-grey-icon' }/>
  },
  {
    id: 3,
    link: '/logout',
    name: 'Sign out',
    icon: <ExitToAppIcon className={ 'light-grey-icon ' }/>
  }
];
