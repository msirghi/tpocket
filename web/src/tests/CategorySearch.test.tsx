import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { CategorySearch } from '../components/category/CategorySearch';
import { findByTestAttr } from './utils/findByTestAttr';
import toJson from 'enzyme-to-json';

describe('CategorySearch Component', () => {
  let component: ShallowWrapper;
  const onChange = jest.fn();
  const onClose = jest.fn();

  const renderComponent = (resultCount: number, resultCountShown: boolean) =>
    shallow(
      <CategorySearch
        onClose={onClose}
        onChange={onChange}
        resultCount={resultCount}
        resultCountShown={resultCountShown}
      />
    );

  beforeEach(() => {
    component = renderComponent(23, true);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render component without errors', () => {
    expect(toJson(renderComponent(23, true))).toMatchSnapshot();
  });

  it('should rendedr right result count', () => {
    const resultCounter = findByTestAttr(component, 'result-counter');
    expect(resultCounter.text()).toBe('23 results found.');
  });

  it('should not render result count on false prop', () => {
    component = renderComponent(12, false);
    const resultCounter = findByTestAttr(component, 'result-counter');
    expect(resultCounter.length).toBe(0);
  });

  it('should not render result if counter is < 0', () => {
    component = renderComponent(-1, true);
    const resultCounter = findByTestAttr(component, 'result-counter');
    expect(resultCounter.length).toBe(0);
  });

  it('should fire method on input change', () => {
    const input = findByTestAttr(component, 'category-name-field');
    input.simulate('change', { target: { value: 'Test value' } });
    expect(onChange).toBeCalled();
  });

  it('should fire onClose prop close button click', () => {
    const button = findByTestAttr(component, 'close-icon');
    button.simulate('click');
    expect(onClose).toBeCalledTimes(1);
  });
});
