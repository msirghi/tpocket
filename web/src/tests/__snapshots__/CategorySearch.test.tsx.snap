// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategorySearch Component should render component without errors 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Grid))
    alignItems="flex-end"
    className="pb-15"
    container={true}
    justify="center"
    spacing={1}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <WithStyles(ForwardRef(TextField))
        data-test="category-name-field"
        label="Category name"
        onChange={[Function]}
        style={
          Object {
            "width": 300,
          }
        }
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <CloseIcon
        data-test="close-icon"
        onClick={[Function]}
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <div
    className="text-center"
  >
    <WithStyles(ForwardRef(Typography))
      data-test="result-counter"
      gutterBottom={true}
      variant="subtitle1"
    >
      23
       results found.
    </WithStyles(ForwardRef(Typography))>
  </div>
</Fragment>
`;
